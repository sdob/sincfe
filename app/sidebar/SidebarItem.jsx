import React from 'react';
import { Link } from 'react-router';

/*
 * A simple presentational component that takes 'icon' and 'text'
 * properties (and, optionally, extra classNames) and returns the
 * HTML we want. It keeps things consistent and saves us boilerplate
 * in the sidebar components.
 */
export default function SidebarItem(props) {
  return (
    <li className={`sinc-sidebar__menu-item ${props.className}`}>
      <Link to={props.to}>
        <i className={`sinc-sidebar__icon fa fa-fw fa-${props.icon}`} />{props.text}
      </Link>
    </li>
  );
}
