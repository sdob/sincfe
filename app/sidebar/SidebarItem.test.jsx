import React from 'react';
import SidebarItem from './SidebarItem';
import renderer from 'react-test-renderer';

test('Some assertion', () => {
  const component = renderer.create(
    <SidebarItem to="/path" text="Content" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
