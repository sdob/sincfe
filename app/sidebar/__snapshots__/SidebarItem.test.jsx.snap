exports[`test Some assertion 1`] = `
<li
  className="sinc-sidebar__menu-item undefined">
  <a
    onClick={[Function]}
    style={Object {}}>
    <i
      className="sinc-sidebar__icon fa fa-fw fa-undefined" />
    Content
  </a>
</li>
`;
